### PR TITLE
shell: fix resize command

### PR DESCRIPTION
**Pull request by jakub-uC**
Resize command requires shell to be working with interrupt
driven API. This will be now enabled by default when shell serial
backend is selected. Recntly this feature has been accidently
switched off: PR #11556.

k_sleep(1) has been replaced with k_busy_wait(1000) to avoid
hanging in a command handler.

Some command help messages have been rephrased in order to decrease total flash usage.

Signed-off-by: Jakub Rzeszutko <jakub.rzeszutko@nordicsemi.no>

### CODE REVIEW

**Review by ceolin — CHANGES_REQUESTED** [review]
Patch looks good, just need to increase the rom requirement to some tests not fail. Also there is one typo in the commit's message.

**Review by nordic-krch — CHANGES_REQUESTED** [review]

File: subsys/shell/Kconfig.backends [hunk 1]
```diff
@@ -27,8 +27,9 @@ if SHELL_BACKEND_SERIAL
 # Internal config to enable UART interrupts if supported.
```
Comment (jukkar) [inline subsys/shell/Kconfig.backends]: Typo in commit msg, s/Recntly/Recently/, better would be to say here "Recently this feature was accidentally turned off in PR #11556"

Comment (jarz-nordic) [reply subsys/shell/Kconfig.backends]: fixed

File: subsys/shell/shell_cmds.c [hunk 2]
```diff
@@ -29,7 +29,7 @@
 	"Reset shell statistics for the Logger module."
 #define SHELL_HELP_RESIZE						\
 	"Console gets terminal screen size or assumes 80 in case "	\
-	"the readout fails. It must be executed after each terminal "	\
+	" the readout fails. It must be executed after each terminal "	\
```
[the remark this answers is not in the document]
Comment (jarz-nordic) [reply subsys/shell/shell_cmds.c]: fixed

File: subsys/shell/shell_cmds.c [hunk 3]
```diff
@@ -8,12 +8,12 @@
...
+#define SHELL_MSG_CMD_NOT_SUPPORTED	"Command not supported.\n"
 #define SHELL_HELP_CLEAR		"Clear screen."
-#define SHELL_HELP_BACKSPACE_MODE	"Toggle backspace key mode.\n" \
-	"Some terminals are not sending separate escape code for"	 \
-	"backspace and delete button. Hence backspace is not working as" \
-	"expected. This command can force shell to interpret delete"	 \
-	" escape as backspace."
+#define SHELL_HELP_BACKSPACE_MODE	"Toggle backspace key mode.\n"	      \
+	"Some terminals are not sending separate escape code for"	      \
+	"backspace and delete button. This command forces shell to interpret" \
```
[the remark this answers is not in the document]
Comment (jarz-nordic) [reply subsys/shell/shell_cmds.c]: fixed

**Comment by jarz-nordic** [comment]
@nordic-krch , @ceolin : I've implemented your findings.

**Review by nordic-krch — APPROVED** [review]

**Review by ceolin — APPROVED** [review]

**Pull request merged**